ddl: unexpected conflict error in parallel ddls

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
/* init */ drop table if exists t, ty;
/* init */ create table t(id int);
/* init */ insert into t values(1), (2);

/* t1 */ begin pessimistic;
/* t1 */ insert into t values(3);
/* t2 */ alter table t add index k(id); -- this will be blocked for about 2 seconds
/* t3 */ create table ty(id int);
/* t1 */ commit;
```

Use [stmtflow](https://github.com/zyguan/sqlz/tree/master/tools/stmtflow) run this SQL file can reproduce.

```sh
./stmtflow play --block-time=1s --dsn="root:@tcp(127.0.0.1:4000)/test?tidb_enable_amend_pessimistic_txn=1" ./conflict.sql
```

### 2. What did you expect to see? (Required)

The DDLs executed successfully without error.

### 3. What did you see instead (Required)

There are conflict error in log.

```log
[2020/11/30 17:02:04.823 +08:00] [ERROR] [ddl_worker.go:157] ["[ddl] handle DDL job failed"] [worker="worker 4, tp add index"] [error="[kv:9007]Write conflict, txnStartTS=421193822319149057, conflictStartTS=421193822528864259, conflictCommitTS=421193822528864261, key=[]byte{0x6d, 0x44, 0x69, 0x66, 0x66, 0x3a, 0x32, 0x39, 0x0, 0xfe, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x73} primary={metaKey=true, key=DB:1, field=Table:49} [try again later]"]
```

### 4. What is your TiDB version? (Required)

```
Release Version: v4.0.0-beta.2-1607-g93c3e6bec
Edition: Community
Git Commit Hash: 93c3e6becd5929cc37901d78d1f679cbc5d86342
Git Branch: master
UTC Build Time: 2020-11-21 13:12:37
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

--- a/ddl/ddl_worker.go
+++ b/ddl/ddl_worker.go
@@ -154,7 +154,7 @@ func (w *worker) start(d *ddlCtx) {
 
 		err := w.handleDDLJobQueue(d)
 		if err != nil {
-			logutil.Logger(w.logCtx).Error("[ddl] handle DDL job failed", zap.Error(err))
+			logutil.Logger(w.logCtx).Warn("[ddl] handle DDL job failed", zap.Error(err))
 		}
 	}
 }

